tests/test_generate_report.py

```diff
@@ -0,0 +1,183 @@
+
+        json_formatted["model_params"]["min_samples_leaf"] = 2
+        f = FinalRecommendationModule(json_formatted)
+        returned = f.process_dict()
+
+        self.assertIn("Min samples per leaf", returned["score_descriptions"][0])
+
+    def test_statistically_significant(self):
+        """test the statistically significant AUC p-values check in FinalRecommendationModule"""
+        json_formatted = self.get_test_report()
+        json_formatted["attack_experiment_logger"]["attack_instance_logger"] = {}
+
+        metrics_dict = {
+            "P_HIGHER_AUC": 0.001,
+            "AUC": 0.8,
+            "ACC": 0.7,
+            "FDIF01": 0.2,
+            "PDIF01": 1.0,
+            "FPR": 1.0,
+            "TPR": 0.1,
+        }
+
+        for i in range(10):
+            json_formatted["attack_experiment_logger"]["attack_instance_logger"][
+                "instance_" + str(i)
+            ] = metrics_dict
+
+        f = FinalRecommendationModule(json_formatted)
+        returned = f.process_dict()
+
+        self.assertIn(
+            ">10% AUC are statistically significant", returned["score_descriptions"][0]
+        )
+        self.assertIn("Attack AUC > threshold", returned["score_descriptions"][1])
+
+    def test_univariate_metrics_module(self):
+        """test the SummariseUnivariateMetricsModule"""
+        json_formatted = self.get_test_report()
+        f = SummariseUnivariateMetricsModule(json_formatted)
+        _ = f.process_dict()
+
+    def test_auc_pvals_module(self):
+        """test the SummariseAUCPvalsModule"""
+        json_formatted = self.get_test_report()
+        f = SummariseAUCPvalsModule(json_formatted)
+        _ = f.process_dict()
+
+        f = SummariseAUCPvalsModule(json_formatted, correction="bo")
+        _ = f.process_dict()
+
+        with pytest.raises(NotImplementedError):
+            f = SummariseAUCPvalsModule(json_formatted, correction="xyzabcd")
+            _ = f.process_dict()
+
+    def test_fdif_pvals_module(self):
+        """test the SummariseFDIFPvalsModule"""
+        json_formatted = self.get_test_report()
+        f = SummariseFDIFPvalsModule(json_formatted)
+        _ = f.process_dict()
+        _ = f.get_metric_list(json_formatted["attack_experiment_logger"])
+
+    def test_loglog_roc_module(self):
+        """test the LogLogROCModule"""
+        json_formatted = self.get_test_report()
+        f = LogLogROCModule(json_formatted)
+        _ = f.process_dict()
+
+        f = LogLogROCModule(json_formatted, output_folder="./")
+        _ = f.process_dict()
+
+    def test_complete_runthrough(self):
+        """test the full process_json file end-to-end when valid parameters are passed"""
+        json_formatted = self.get_test_report()
+        _ = self.process_json_from_file(json_formatted)
+
+    def test_cleanup(self):
+        """gets rid of files created during tests"""
+        names = ["test.json", "results.txt", "1024-WorstCase attack.png"]
```

Comment:
this might be my fault, but could we fix the module that plots to not produce filenames with whitespaces?

Reply:
Created issue #130 to cover this